File: java/jedis/src/main/java/com/rl/sample/jedis/Sentinel.java
package com.rl.sample.jedis;

import redis.clients.jedis.Jedis;
import redis.clients.jedis.JedisSentinelPool;

import java.util.HashSet;

public class Sentinel {
    String host;
    int port;
    String password;
    String service;


    public static void main(String[] args) {
        if (!(args.length == 3 || args.length == 4) ) {
            System.out.println("Usage: sentinelhost sentinelport service password");
            System.exit(1);
        }

        Sentinel example = new Sentinel();
        example.host = args[0];
        example.port = Integer.parseInt(args[1]);
        example.service = args[2];
        if (args.length == 4) example.password = args[3];

        JedisSentinelPool pool = example.getPool();

        try (Jedis jedis = pool.getResource()) {
            String resp = jedis.set("foo", "bar");
            System.out.println("Set: " + resp);
            System.out.println("Get: " + jedis.get("foo"));
        }

        //close the pool when application terminates
        pool.close();
    }


    private JedisSentinelPool getPool() {
        HashSet sentinels = new HashSet<String>();
        //You can add more sentinels here
        sentinels.add(host + ":" + port);
        return new JedisSentinelPool(service, sentinels, password);
    }



}

File: python/sentinel.py
import redis
import redis.sentinel
import sys

def main():
    if not(len(sys.argv) == 4 or len(sys.argv) == 5):
        print("Usage: sentinelhost sentinelport service password")
        exit(1)
    sentinel_host = sys.argv[1]
    sentinel_port = sys.argv[2]
    service_name = sys.argv[3]

    password = None
    if len(sys.argv) == 5:
        password = sys.argv[4]

    #you can add more sentinels to the list 
    sentinel_list = [(sentinel_host,sentinel_port)]
    s = redis.sentinel.Sentinel(sentinel_list, socket_timeout=0.1)
    r = s.master_for(service_name, socket_timeout=0.1, password=password)
    
    print("Connected to Redis")
    print("Set: {}".format (r.set("foo","bar")))
    print("Get: {}".format(r.get("foo")))
    
    #just for completeness close the connection pool
    r.connection_pool.disconnect()
    

if __name__== "__main__":
  main()